Src/FluentAssertions/Equivalency/Steps/GenericDictionaryEquivalencyStep.cs

```diff
@@ -20,10 +21,17 @@ public class GenericDictionaryEquivalencyStep : IEquivalencyStep
         if (comparands.Expectation != null)
         {
             Type expectationType = comparands.GetExpectedType(context.Options);
+
             if (DictionaryInterfaceInfo.FindFrom(expectationType, "expectation") is { } expectedDictionary)
             {
+                if (SubjectIsNonGenericDictionary(comparands.Subject))
```

Comment:
🤔 Can't you move this before everything else as an early exit?

Reply:
No, unfortunately not, because `DictinaryInterfaceInfo.FindFrom` guard for dictionaries which implements multiple `IDictionary` and if these are non-generic we will move to the `DictionaryEquivalencyStep` immediately and therefore we do not fail in such a case.

Reply:
Of course we could combine the `if` with `||` but why? 😄